include/net/golioth.h

```diff
@@ -64,8 +82,11 @@ struct golioth_client {
 	int sock;
 
 	void (*on_connect)(struct golioth_client *client);
-	void (*on_message)(struct golioth_client *client,
-			   struct coap_packet *rx);
+	void (*on_message)(struct golioth_client *client, struct coap_packet *rx);
+
+	/* Storage for additional on_message callbacks */
+	struct golioth_message_callback_reg message_callbacks[GOLIOTH_MAX_NUM_MESSAGE_CALLBACKS];
+	size_t num_message_callbacks;
```

Comment:
We willy rework that and hide `on_message()` with the API rework. But for now it looks reasonable the way it is here.